mobile/status.go

```diff
@@ -0,0 +1,338 @@
+	return makeJSONResponse(err)
+}
+
+//Login loads a key file (for a given address), tries to decrypt it using the password, to verify ownership
+// if verified, purges all the previous identities from Whisper, and injects verified key as shh identity
+func Login(address, password string) string {
+	err := statusBackend.SelectAccount(address, password)
+	return makeJSONResponse(err)
+}
+
+//Logout is equivalent to clearing whisper identities
+func Logout() string {
+	err := statusBackend.Logout()
+	return makeJSONResponse(err)
+}
+
+// SignMessage unmarshals rpc params {data, address, password} and passes
+// them onto backend.SignMessage
+func SignMessage(rpcParams string) string {
+	var params personal.SignParams
+	err := json.Unmarshal([]byte(rpcParams), &params)
+	if err != nil {
+		return prepareJSONResponseWithCode(nil, err, codeFailedParseParams)
+	}
+	result, err := statusBackend.SignMessage(params)
+	return prepareJSONResponse(result.String(), err)
+}
+
+// Recover unmarshals rpc params {signDataString, signedData} and passes
+// them onto backend.
+func Recover(rpcParams string) string {
+	var params personal.RecoverParams
+	err := json.Unmarshal([]byte(rpcParams), &params)
+	if err != nil {
+		return prepareJSONResponseWithCode(nil, err, codeFailedParseParams)
+	}
+	addr, err := statusBackend.Recover(params)
+	return prepareJSONResponse(addr.String(), err)
+}
+
+// SendTransaction converts RPC args and calls backend.SendTransaction
+func SendTransaction(txArgsJSON, password string) string {
+	var params transactions.SendTxArgs
+	err := json.Unmarshal([]byte(txArgsJSON), &params)
+	if err != nil {
+		return prepareJSONResponseWithCode(nil, err, codeFailedParseParams)
+	}
+	hash, err := statusBackend.SendTransaction(params, password)
+	code := codeUnknown
+	if c, ok := errToCodeMap[err]; ok {
+		code = c
+	}
+	return prepareJSONResponseWithCode(hash.String(), err, code)
+}
+
+//StartCPUProfile runs pprof for cpu
+func StartCPUProfile(dataDir string) string {
+	err := profiling.StartCPUProfile(dataDir)
+	return makeJSONResponse(err)
+}
+
+//StopCPUProfiling stops pprof for cpu
+func StopCPUProfiling() string { //nolint: deadcode
+	err := profiling.StopCPUProfile()
+	return makeJSONResponse(err)
+}
+
+//WriteHeapProfile starts pprof for heap
+func WriteHeapProfile(dataDir string) string { //nolint: deadcode
+	err := profiling.WriteHeapFile(dataDir)
+	return makeJSONResponse(err)
+}
+
+func makeJSONResponse(err error) string {
+	errString := ""
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		errString = err.Error()
+	}
+
+	out := APIResponse{
+		Error: errString,
+	}
+	outBytes, _ := json.Marshal(out)
+
+	return string(outBytes)
+}
+
+// NotifyUsers sends push notifications by given tokens.
+func NotifyUsers(message, payloadJSON, tokensArray string) (outCBytes string) {
+	var (
+		err      error
+		outBytes []byte
+	)
+	errString := ""
+
+	defer func() {
```

Comment:
this defer usage is so confusing. it definitely should be a simple function that accepts error and returns string